src/main/java/ch/uzh/ifi/hase/soprafs23/service/UserService.java

```diff
@@ -140,9 +142,11 @@ public void updateUser(Long id, UserPutDTO userPutDTO,String currentPassword) {
         }
 
         if(newAllergies != null) {
-            user.setAllergies(newAllergies);
+            Set<String> allergiesSet = new HashSet<>(newAllergies); //convert here from List to Set to not make it in UserPutDTO //TODO: not necessary anymore?
```

Comment:
as the TODO says, I don't think this is necessary as newAllergies is already a set not a list

Reply:
jep, cleaned it up